frontend/src/containers/WidgetWrapper.js

```diff
@@ -0,0 +1,137 @@
+import React, { PureComponent } from 'react';
+import PropTypes from 'prop-types';
+import { connect } from 'react-redux';
+
+import {
+  openModal,
+  patch,
+  updatePropertyValue,
+  allowShortcut,
+  disableShortcut,
+} from '../actions/WindowActions';
+import { getCellWidgetData } from '../utils/tableHelpers';
+import { getTable } from '../reducers/tables';
+import {
+  getMasterData,
+  getMasterWidgetData,
+  getMasterWidgetFields,
+  getMasterDocStatus,
+} from '../reducers/windowHandler';
+
+import MasterWidget from '../components/widget/MasterWidget';
+import RawWidget from '../components/widget/RawWidget';
+
+/**
+ * @file Class based component.
+ * @module WidgetWrapper
+ * @extends PureComponent
+ */
```

Comment:
I believe you could write here some docs explaining briefly how this is functioning since that would be the central logic component. Thanks